daemon/cluster/executor/container/container.go

```diff
@@ -207,7 +206,7 @@ func getMountMask(m *api.Mount) string {
 func (c *containerConfig) hostConfig() *enginecontainer.HostConfig {
 	return &enginecontainer.HostConfig{
 		Resources: c.resources(),
-		Binds:     c.bindMounts(),
+		Binds:     append(c.bindMounts(), c.volumes()...),
```

Comment:
Ok... one thing I don't like here (different topic) is docker will automatically create the volume with the given name if it doesn't already exist.

For example:
```
$ docker service create --mount type=volume,source=test,target=/foo busybox top
```
Even though we've not provided any create options, we'll get a volume with the specified name, on the local driver.